src/components/TokenEnterAmount.test.tsx

```diff
@@ -79,141 +55,421 @@ describe('TokenEnterAmount', () => {
-  })
+  describe.each([
+    { decimal: '.', group: ',' },
+    { decimal: ',', group: '.' },
+  ])('with decimal separator "$decimal" and group separator "$group"', ({ decimal, group }) => {
+    beforeEach(() => {
+      jest.mocked(getNumberFormatSettings).mockReturnValue({
+        decimalSeparator: decimal,
+        groupingSeparator: group,
+      })
+    })
 
-  it('displays the correct token information', () => {
-    const store = createMockStore(mockStore)
-    const { getByText, getByTestId } = render(
-      <Provider store={store}>
-        <TokenEnterAmount {...defaultProps} />
-      </Provider>
-    )
-    expect(getByTestId('tokenEnterAmount/TokenName')).toBeTruthy()
-    expect(getByText('CELO on Celo Alfajores')).toBeTruthy()
-    expect(getByTestId('tokenEnterAmount/SwitchTokens')).toBeTruthy()
-    expect(getByTestId('tokenEnterAmount/TokenSelect')).toBeTruthy()
-    expect(getByTestId('tokenEnterAmount/TokenBalance')).toBeTruthy()
-    expect(getByTestId('tokenEnterAmount/TokenBalance').props.children.props.i18nKey).toBe(
-      'tokenEnterAmount.availableBalance'
-    )
-  })
+    const replaceSeparators = (value: string) =>
+      value.replace(/\./g, '|').replace(/,/g, group).replace(/\|/g, decimal)
 
-  it('formats the input value correctly', () => {
-    const store = createMockStore(mockStore)
-    const { getByTestId, rerender } = render(
-      <Provider store={store}>
-        <TokenEnterAmount {...defaultProps} />
-      </Provider>
-    )
-    const input = getByTestId('tokenEnterAmount/TokenAmountInput')
-    const converted = getByTestId('tokenEnterAmount/ExchangeAmount')
-    expect(input.props.value).toBe('1')
-    expect(converted.props.children).toBe(`${APPROX_SYMBOL} $0.1`)
-    fireEvent.press(getByTestId('tokenEnterAmount/SwitchTokens'))
-
-    // simulate call of toggleAmountType
-    rerender(
-      <Provider store={store}>
-        <TokenEnterAmount
-          {...defaultProps}
-          amountType="local"
-          inputValue="0.1"
-          tokenAmount="1 CELO"
-        />
-      </Provider>
-    )
-
-    expect(input.props.value).toBe('₱0.1')
-    expect(converted.props.children).toBe(`${APPROX_SYMBOL} 1 CELO`)
-  })
+    it('properly groups numbers', () => {
+      expect(formatNumber('')).toBe('')
+      expect(formatNumber('123')).toBe(replaceSeparators('123'))
+      expect(formatNumber('1234')).toBe(replaceSeparators('1,234'))
+      expect(formatNumber('1234567')).toBe(replaceSeparators('1,234,567'))
+      expect(formatNumber('1234567.12345')).toBe(replaceSeparators('1,234,567.12345'))
+      expect(formatNumber('123456789012345')).toBe(replaceSeparators('123,456,789,012,345'))
+      expect(formatNumber('12.34567')).toBe(replaceSeparators('12.34567'))
+      expect(formatNumber('-1234567.89')).toBe(replaceSeparators('-1,234,567.89'))
+      expect(formatNumber('1234abc')).toBe(replaceSeparators('1,234abc'))
+      expect(formatNumber('1234.56xyz')).toBe(replaceSeparators('1,234.56xyz'))
```

Comment:
nit: i think 1234 and 1234567 and 123456789012345 and -1234567.89 essentially all test for the grouping logic - perhaps we can consider consolidating some them into one test case (same with the tests below)? maybe some  also wondering if we need to cater for the cases with the letters since there should be no letters allowed?

Reply:
@kathaypacific 1. I just thought that it might be helpful to have each dedicated test case for each function (probably, should've named the tests accordingly then) but I don't have a strong stance on this so either approach works for me!

Reply:
ahh i see, up to you if you want to leave the tests with the letters. if we think about tests as documentation and a "spec" for the component, and during a refactor if these tests no longer pass, should that warrant extra work to get them to pass (i.e. do we care about this scenario, to document it to ensure it always works?)